include/crm/common/xml_internal.h

```diff
@@ -190,6 +190,14 @@ int pcmk__xml_show_changes(pcmk__output_t *out, const xmlNode *xml);
 #define PCMK__XP_REMOTE_NODE_STATUS \
     "//" XML_TAG_CIB "//" XML_CIB_TAG_STATUS "//" XML_CIB_TAG_STATE \
     "[@" XML_NODE_IS_REMOTE "='true']"
+/*!
+ * \internal
+ * \brief Serialize XML (using libxml) into provided descriptor
+ *
+ * \param[in] fd  File descriptor to (piece-wise) write to
+ * \param[in] cur XML subtree to proceed
+ */
+void pcmk__xml_serialize_fd_formatted(int fd, xmlNode *cur);
```

Comment:
FYI, our various `dump` functions help convert an `xmlNode*` to a textual string (for display, logging, writing to a file, etc.). Since poki's original PR, we've established a precedent by renaming one of them to `pcmk__xml2text()` for readability, so I'm thinking we can extend that here by naming this `pcmk__xml2fd()`.

We have our own `dump` functions instead of simply using the libxml2 APIs that this new function calls because we implement some extensions (`pcmk__xml_fmt_options`), and apparently at least in the past the libxml2 functions were horrendously slow. However the libxml2 functions should be much faster when writing directly to a file descriptor, because it avoids the overhead of converting the entire XML to a string and then writing out that string. The goal of this new function is to use the libxml2 functions when we're writing to a file and don't need any `pcmk__xml_fmt_options`.